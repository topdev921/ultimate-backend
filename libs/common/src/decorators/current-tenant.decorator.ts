import { createParamDecorator } from '@nestjs/common';

export const CurrentTenant = createParamDecorator(
  (data, [root, args, ctx, info]) => ctx.req.tenant,
);
