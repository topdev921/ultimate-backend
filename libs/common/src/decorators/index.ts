export * from './current-user.decorator';
export * from './current-tenant.decorator';
