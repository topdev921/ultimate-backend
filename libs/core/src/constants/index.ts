export * from './services.constant';
export * from './service.constants';
