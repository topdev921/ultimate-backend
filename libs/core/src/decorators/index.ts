export * from './roles.decorator';
export * from './filterable';
export * from './permission.decorator';
export * from './resource.decorator';
export * from './indentity-params.decorators';
