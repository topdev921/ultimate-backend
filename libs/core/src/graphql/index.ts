export * from './extensions/filed-roles.guard';
export * from './plugins/query-complexity.plugin';
