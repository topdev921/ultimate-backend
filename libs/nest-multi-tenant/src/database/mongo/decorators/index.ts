export * from './mongo.decorator';
export * from './entity-repository.decorator';
export * from './hooks.decorator';
export * from './entity.decorator';
export * from './object-id-column.decorator';
