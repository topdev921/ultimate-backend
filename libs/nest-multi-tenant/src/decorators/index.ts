export * from './hooks.decorator';
export * from './entity.decorator';
export * from './object-id-column.decorator';
