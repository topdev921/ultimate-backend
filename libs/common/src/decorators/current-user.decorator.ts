import { createParamDecorator, ExecutionContext } from '@nestjs/common';
import { GqlExecutionContext } from '@nestjs/graphql';

export const CurrentUser = createParamDecorator(
  (data: unknown, context: ExecutionContext) => {
    const ctx = GqlExecutionContext.create(context).getContext();
    return ctx.req.user;
  },
);

export const GrpcCurrentUser = createParamDecorator((data, { args }) => {
  return args?.currentUser;
});
