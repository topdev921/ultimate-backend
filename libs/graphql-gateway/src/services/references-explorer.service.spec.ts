/* tslint:disable:max-classes-per-file */
import { INestApplicationContext, Module } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';
import { MetadataScanner } from '@nestjs/core/metadata-scanner';
import { Resolver } from '@nestjs/graphql';
import { GRAPHQL_MODULE_OPTIONS } from '@nestjs/graphql/dist/graphql.constants';
import { ResolveReference } from '../decorators';
import { ReferencesExplorerService } from './references-explorer.service';

@Resolver()
class AccountsResolver {
  @Resolver('User')
  @ResolveReference()
  // tslint:disable-next-line
  public user() {}
}

@Module({
  providers: [
    {
      provide: GRAPHQL_MODULE_OPTIONS,
      useValue: {},
    },
    AccountsResolver,
    MetadataScanner,
    ReferencesExplorerService,
  ],
})
class AppModule {}

describe('ReferencesExplorerService', () => {
  // tslint:disable-next-line:prefer-const
  let app: INestApplicationContext;
  // tslint:disable-next-line:prefer-const
  let referencesExplorer: ReferencesExplorerService;

  beforeEach(async () => {
    app = await NestFactory.createApplicationContext(AppModule);
    referencesExplorer = app.get(ReferencesExplorerService);
  });

  describe('explore', () => {
    it('should return User with __resolveReference', () => {
      const resolvers = referencesExplorer.explore();

      expect(resolvers).toEqual({
        User: {
          __resolveReference: expect.any(Function),
        },
      });
      expect(resolvers).toMatchSnapshot();
    });
  });
});
