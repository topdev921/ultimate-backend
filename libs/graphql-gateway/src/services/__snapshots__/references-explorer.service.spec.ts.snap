// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferencesExplorerService explore should return User with __resolveReference 1`] = `
Object {
  "User": Object {
    "__resolveReference": [Function],
  },
}
`;
